pkg/odo/cli/component/create.go

```diff
@@ -38,280 +41,276 @@ var (
 
-  # Create new Wildfly component with binary named sample.war in './downloads' directory
-  odo create wildfly wildly --binary ./downloads/sample.war
 
-  # Create new Node.js component with the source in current directory and ports 8080-tcp,8100-tcp and 9100-udp exposed
-  odo create nodejs --port 8080,8100/tcp,9100/udp
+# Create a new Node.js component of version 6 from the 'openshift' namespace
+odo create openshift/nodejs:6 --local /nodejs-ex
 
-  # Create new Node.js component with the source in current directory and env variables key=value and key1=value1 exposed
-  odo create nodejs --env key=value,key1=value1
+# Create new Wildfly component with binary named sample.war in './downloads' directory
+odo create wildfly wildly --binary ./downloads/sample.war
 
-  # For more examples, visit: https://github.com/redhat-developer/odo/blob/master/docs/examples.md
-  odo create python --git https://github.com/openshift/django-ex.git
-	`,
-	Args: cobra.RangeArgs(1, 2),
-	Run: func(cmd *cobra.Command, args []string) {
+# Create new Node.js component with the source in current directory and ports 8080-tcp,8100-tcp and 9100-udp exposed
+odo create nodejs --port 8080,8100/tcp,9100/udp
 
-		stdout := color.Output
-		glog.V(4).Infof("Component create called with args: %#v, flags: binary=%s, git=%s, local=%s", strings.Join(args, " "), componentBinary, componentGit, componentLocal)
+# Create new Node.js component with the source in current directory and env variables key=value and key1=value1 exposed
+odo create nodejs --env key=value,key1=value1
 
-		context := genericclioptions.NewContextCreatingAppIfNeeded(cmd)
-		client := context.Client
-		projectName := context.Project
-		applicationName := context.Application
+# For more examples, visit: https://github.com/redhat-developer/odo/blob/master/docs/examples.md
+odo create python --git https://github.com/openshift/django-ex.git
+  `)
 
-		checkFlag := 0
-		componentPath := ""
-		var componentPathType occlient.CreateType
+// NewCreateOptions returns new instance of CreateOptions
+func NewCreateOptions() *CreateOptions {
+	return &CreateOptions{}
+}
 
-		if len(componentBinary) != 0 {
-			componentPath = componentBinary
-			path, err := util.GetAbsPath(componentPath)
-			odoutil.LogErrorAndExit(err, "Failed to resolve %s to absolute path", componentPath)
-			componentPath = path
-			componentPathType = occlient.BINARY
-			checkFlag++
-		}
-		if len(componentGit) != 0 {
-			componentPath = componentGit
-			componentPathType = occlient.GIT
-			checkFlag++
-		}
-		if len(componentLocal) != 0 {
-			componentPath = componentLocal
-			path, err := util.GetAbsPath(componentPath)
-			odoutil.LogErrorAndExit(err, "Failed to resolve %s to absolute path", componentPath)
-			componentPath = path
-			componentPathType = occlient.LOCAL
-			checkFlag++
-		}
+// Complete completes create args
+func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
```

Comment:
> I think, in this case, that the validation that only one of `source`, `local` or `binary` should occur here to avoid useless processing and fail as quickly as possible.

Ack Done